graphql-java-spring-webflux/src/main/java/graphql/spring/web/reactive/components/GraphQLController.java

```diff
@@ -47,6 +52,14 @@ public Object graphqlPOST(
             @RequestBody(required = false) String body,
             ServerWebExchange serverWebExchange) {
 
+        MediaType mediaType = null;
+        if (!StringUtils.isEmpty(contentType)) {
+            try {
+                mediaType = MediaType.parseMediaType(contentType);
+            } catch (InvalidMediaTypeException ignore) {
```

Comment:
why are we ignoring wrong content types here? is that a real case we need to deal with?

Reply:
If the parsing fails at that point we could throw also a `throw new ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, "Could not process GraphQL request");`

Reply:
Note we do the same at the end in that method. Do you prefer to add that?

You want a test case with an invalid content type? I can add that one yes.

Reply:
@andimarek I can do something like `throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid Content-Type: " + contentType, e);`

Testing will be difficult as MockMvc does not allow to send invalid media types:
https://github.com/spring-projects/spring-framework/blob/master/spring-test/src/main/java/org/springframework/test/web/servlet/request/MockHttpServletRequestBuilder.java#L648

So effectively the test will never get in the controller itself (I just tried).

Let me know if you prefer the 400 I suggested above or keep as it and fail later (HTTP 422 at end of controller)?